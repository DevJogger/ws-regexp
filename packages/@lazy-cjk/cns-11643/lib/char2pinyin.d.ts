import { EnumPinyinMode, IZhuyin2PinyinTableRow, IZhuyin2PinyinTable, EnumPinyinType } from './types';
export declare function zhuyin2pinyin_table(pinyinMode?: EnumPinyinMode): IZhuyin2PinyinTable;
export declare function zhuyin2pinyin(zhuyin: string, pinyinMode?: EnumPinyinMode): IZhuyin2PinyinTableRow;
export declare function uni2pinyin(uni: string | number, pinyinMode?: EnumPinyinMode): IZhuyin2PinyinTableRow;
export declare function uni2pinyinTypeValue(uni: string | number, pinyinMode?: EnumPinyinMode, pinyinType?: EnumPinyinType): string;
export declare function char2pinyin(char: string, pinyinMode?: EnumPinyinMode): IZhuyin2PinyinTableRow;
export declare function char2pinyinTypeValue(char: string, pinyinMode?: EnumPinyinMode, pinyinType?: EnumPinyinType): string;
export default char2pinyin;
