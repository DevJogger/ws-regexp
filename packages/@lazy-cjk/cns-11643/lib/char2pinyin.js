"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.char2pinyinTypeValue = exports.char2pinyin = exports.uni2pinyinTypeValue = exports.uni2pinyin = exports.zhuyin2pinyin = exports.zhuyin2pinyin_table = void 0;
const uni2pinyin_01_1 = __importStar(require("./uni2pinyin_01"));
const uni2pinyin_02_1 = __importStar(require("./uni2pinyin_02"));
const uni2zhuyin_1 = require("./util/uni2zhuyin");
function zhuyin2pinyin_table(pinyinMode) {
    const zhuyin2pinyinMode = pinyinMode ? uni2pinyin_01_1.zhuyin2pinyin_01_table : uni2pinyin_02_1.zhuyin2pinyin_02_table;
    return zhuyin2pinyinMode();
}
exports.zhuyin2pinyin_table = zhuyin2pinyin_table;
function zhuyin2pinyin(zhuyin, pinyinMode) {
    const zhuyin2pinyinMode = pinyinMode ? uni2pinyin_01_1.zhuyin2pinyin_01 : uni2pinyin_02_1.zhuyin2pinyin_02;
    return zhuyin2pinyinMode(zhuyin);
}
exports.zhuyin2pinyin = zhuyin2pinyin;
function uni2pinyin(uni, pinyinMode) {
    const uni2pinyinMode = pinyinMode ? uni2pinyin_01_1.default : uni2pinyin_02_1.default;
    return uni2pinyinMode(uni);
}
exports.uni2pinyin = uni2pinyin;
function uni2pinyinTypeValue(uni, pinyinMode, pinyinType) {
    pinyinType = uni2zhuyin_1.handlePinyinType(pinyinType);
    return uni2pinyin(uni, pinyinMode)[pinyinType];
}
exports.uni2pinyinTypeValue = uni2pinyinTypeValue;
function char2pinyin(char, pinyinMode) {
    const char2pinyinMode = pinyinMode ? uni2pinyin_01_1.char2pinyin_01 : uni2pinyin_02_1.char2pinyin_02;
    return char2pinyinMode(char);
}
exports.char2pinyin = char2pinyin;
function char2pinyinTypeValue(char, pinyinMode, pinyinType) {
    pinyinType = uni2zhuyin_1.handlePinyinType(pinyinType);
    return char2pinyin(char, pinyinMode)[pinyinType];
}
exports.char2pinyinTypeValue = char2pinyinTypeValue;
exports.default = char2pinyin;
//# sourceMappingURL=char2pinyin.js.map