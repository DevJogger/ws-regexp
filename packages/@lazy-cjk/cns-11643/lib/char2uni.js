"use strict";
/**
 * Created by user on 2020/5/30.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.char2uni = char2uni;
exports.uni2char = uni2char;
exports.char2hex = char2hex;
exports.hex2char = hex2char;
exports.uni2hex = uni2hex;
function char2uni(char) {
    return char.codePointAt(0);
}
function uni2char(uni) {
    return String.fromCodePoint(uni);
}
function char2hex(char) {
    return char2uni(char).toString(16);
}
function hex2char(hex) {
    return uni2char(parseInt(hex, 16));
}
function uni2hex(uni) {
    uni = parseInt(uni.toString());
    return uni.toString(16);
}
//# sourceMappingURL=char2uni.js.map