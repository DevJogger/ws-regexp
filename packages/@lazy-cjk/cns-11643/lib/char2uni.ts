/**
 * Created by user on 2020/5/30.
 */

export function char2uni(char: string)
{
	return char.codePointAt(0)
}

export function uni2char(uni: string | number)
{
	return String.fromCodePoint(uni as number)
}

export function char2hex(char: string)
{
	return char2uni(char).toString(16)
}

export function hex2char(hex: string)
{
	return uni2char(parseInt(hex, 16))
}

export function uni2hex(uni: string | number)
{
	uni = parseInt(uni.toString())
	return uni.toString(16)
}

