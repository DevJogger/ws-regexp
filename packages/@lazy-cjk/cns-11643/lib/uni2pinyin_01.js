"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.char2pinyinTypeValue_01 = exports.char2pinyin_01 = exports.uni2pinyinTypeValue_01 = exports.uni2pinyin_01 = exports.zhuyin2pinyin_01 = exports.zhuyin2pinyin_01_table = void 0;
const tslib_1 = require("tslib");
/**
 * Created by user on 2020/5/30.
 */
const uni2zhuyin_1 = tslib_1.__importDefault(require("./uni2zhuyin"));
const char2uni_1 = require("./char2uni");
const uni2zhuyin_2 = require("./util/uni2zhuyin");
/**
 * 全字庫的拼音資料表格
 *
 * 「CNS_pinyin_1」以調值(數字)呈現
 *
 * 第一個欄位：注音
 * 第二個欄位：漢語(han)
 * 第三個欄位：注音第二式(zuin2)
 * 第四個欄位：耶魯(yale)
 * 第五個欄位：韋式(wei)
 */
function zhuyin2pinyin_01_table() {
    return require('./cns/pinyin/pinyin_01.json');
}
exports.zhuyin2pinyin_01_table = zhuyin2pinyin_01_table;
function zhuyin2pinyin_01(zhuyin) {
    return zhuyin2pinyin_01_table()[zhuyin];
}
exports.zhuyin2pinyin_01 = zhuyin2pinyin_01;
/**
 * 全字庫的拼音資料表格
 *
 * 「CNS_pinyin_1」以調值(數字)呈現
 */
function uni2pinyin_01(uni) {
    let zhuyin = uni2zhuyin_1.default(uni);
    return zhuyin2pinyin_01_table()[zhuyin];
}
exports.uni2pinyin_01 = uni2pinyin_01;
function uni2pinyinTypeValue_01(uni, pinyinType) {
    pinyinType = uni2zhuyin_2.handlePinyinType(pinyinType);
    return uni2pinyin_01(uni)[pinyinType];
}
exports.uni2pinyinTypeValue_01 = uni2pinyinTypeValue_01;
/**
 * 全字庫的拼音資料表格
 *
 * 「CNS_pinyin_1」以調值(數字)呈現
 */
function char2pinyin_01(char) {
    return uni2pinyin_01(char2uni_1.char2uni(char));
}
exports.char2pinyin_01 = char2pinyin_01;
function char2pinyinTypeValue_01(char, pinyinType) {
    pinyinType = uni2zhuyin_2.handlePinyinType(pinyinType);
    return char2pinyin_01(char)[pinyinType];
}
exports.char2pinyinTypeValue_01 = char2pinyinTypeValue_01;
exports.default = uni2pinyin_01;
//# sourceMappingURL=uni2pinyin_01.js.map