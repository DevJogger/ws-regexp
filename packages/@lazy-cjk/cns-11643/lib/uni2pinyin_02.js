"use strict";
/**
 * Created by user on 2020/5/30.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.char2pinyinTypeValue_02 = exports.char2pinyin_02 = exports.uni2pinyinTypeValue_02 = exports.uni2pinyin_02 = exports.zhuyin2pinyin_02 = exports.zhuyin2pinyin_02_table = void 0;
const tslib_1 = require("tslib");
const uni2zhuyin_1 = tslib_1.__importDefault(require("./uni2zhuyin"));
const char2uni_1 = require("./char2uni");
const uni2zhuyin_2 = require("./util/uni2zhuyin");
/**
 * 全字庫的拼音資料表格
 *
 * 「CNS_pinyin_2」以聲調符號呈現
 *
 * 第一個欄位：注音
 * 第二個欄位：漢語(han)
 * 第三個欄位：注音第二式(zuin2)
 * 第四個欄位：耶魯(yale)
 * 第五個欄位：韋式(wei)
 */
function zhuyin2pinyin_02_table() {
    return require('./cns/pinyin/pinyin_02.json');
}
exports.zhuyin2pinyin_02_table = zhuyin2pinyin_02_table;
function zhuyin2pinyin_02(zhuyin) {
    return zhuyin2pinyin_02_table()[zhuyin];
}
exports.zhuyin2pinyin_02 = zhuyin2pinyin_02;
/**
 * 全字庫的拼音資料表格
 *
 * 「CNS_pinyin_2」以聲調符號呈現
 */
function uni2pinyin_02(uni) {
    let zhuyin = uni2zhuyin_1.default(uni);
    return zhuyin2pinyin_02_table()[zhuyin];
}
exports.uni2pinyin_02 = uni2pinyin_02;
function uni2pinyinTypeValue_02(uni, pinyinType) {
    pinyinType = uni2zhuyin_2.handlePinyinType(pinyinType);
    return uni2pinyin_02(uni)[pinyinType];
}
exports.uni2pinyinTypeValue_02 = uni2pinyinTypeValue_02;
/**
 * 全字庫的拼音資料表格
 *
 * 「CNS_pinyin_2」以聲調符號呈現
 */
function char2pinyin_02(char) {
    return uni2pinyin_02(char2uni_1.char2uni(char));
}
exports.char2pinyin_02 = char2pinyin_02;
function char2pinyinTypeValue_02(char, pinyinType) {
    pinyinType = uni2zhuyin_2.handlePinyinType(pinyinType);
    return char2pinyin_02(char)[pinyinType];
}
exports.char2pinyinTypeValue_02 = char2pinyinTypeValue_02;
exports.default = uni2pinyin_02;
//# sourceMappingURL=uni2pinyin_02.js.map