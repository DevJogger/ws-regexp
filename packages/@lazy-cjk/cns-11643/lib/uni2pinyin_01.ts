/**
 * Created by user on 2020/5/30.
 */
import uni2zhuyin from './uni2zhuyin';
import { ICNSPinyinTable, ICNSPinyinTableRow } from './types';
import { char2uni } from './char2uni';

export function uni2pinyin_01(uni: string | number): ICNSPinyinTableRow
{
	let zhuyin = uni2zhuyin(uni)

	return (require('./cns/pinyin/pinyin_01.json') as any as ICNSPinyinTable)[zhuyin]
}

export function char2pinyin_01(char: string)
{
	return uni2pinyin_01(char2uni(char))
}

export default uni2pinyin_01
