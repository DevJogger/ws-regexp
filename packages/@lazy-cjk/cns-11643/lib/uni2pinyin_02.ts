/**
 * Created by user on 2020/5/30.
 */

import uni2zhuyin from './uni2zhuyin';
import { IZhuyin2PinyinTable, IZhuyin2PinyinTableRow } from './types';
import { char2uni } from './char2uni';

/**
 * 全字庫的拼音資料表格
 *
 * 「CNS_pinyin_2」以聲調符號呈現
 *
 * 第一個欄位：注音
 * 第二個欄位：漢語(han)
 * 第三個欄位：注音第二式(zuin2)
 * 第四個欄位：耶魯(yale)
 * 第五個欄位：韋式(wei)
 */
export function zhuyin2pinyin_02_table(): IZhuyin2PinyinTable
{
	return require('./cns/pinyin/pinyin_02.json') as any
}

/**
 * 全字庫的拼音資料表格
 *
 * 「CNS_pinyin_2」以聲調符號呈現
 */
export function uni2pinyin_02(uni: string | number): IZhuyin2PinyinTableRow
{
	let zhuyin = uni2zhuyin(uni)

	return zhuyin2pinyin_02_table()[zhuyin]
}

/**
 * 全字庫的拼音資料表格
 *
 * 「CNS_pinyin_2」以聲調符號呈現
 */
export function char2pinyin_02(char: string)
{
	return uni2pinyin_02(char2uni(char))
}

export default uni2pinyin_02
