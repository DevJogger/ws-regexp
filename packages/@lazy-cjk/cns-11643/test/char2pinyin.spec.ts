import char_list from './char_list';
import { char2uni, char2hex, uni2char, hex2char } from '../lib/char2uni';
import { char2pinyin_02 } from '../lib/uni2pinyin_02';
import { char2pinyin_01 } from '../lib/uni2pinyin_01';

describe(`char2pinyin_01`, () =>
{

	char_list
		.forEach(char =>
		{

			test(char, () =>
			{

				let arr = char2pinyin_01(char);

				expect(arr).toBeInstanceOf(Array);
				expect(arr).toMatchSnapshot();

			});

		})
	;

})

describe(`char2pinyin_02`, () =>
{

	char_list
		.forEach(char =>
		{

			test(char, () =>
			{

				let arr = char2pinyin_02(char);

				expect(arr).toBeInstanceOf(Array);
				expect(arr).toMatchSnapshot();

			});

		})
	;

})
