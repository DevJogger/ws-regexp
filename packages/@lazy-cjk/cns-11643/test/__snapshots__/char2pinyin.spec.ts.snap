// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`char2pinyin_01 𠬠 1`] = `
[
  "mou3",
  "mou3",
  "mou3",
  "mou3",
]
`;

exports[`char2pinyin_01 𠮷 1`] = `
[
  "ji2",
  "ji2",
  "ji2",
  "chi2",
]
`;

exports[`char2pinyin_01 𡬶 1`] = `
[
  "xin2",
  "shin2",
  "syin2",
  "hsin2",
]
`;

exports[`char2pinyin_01 𢎐 1`] = `
[
  "shang4",
  "shang4",
  "shang4",
  "shang4",
]
`;

exports[`char2pinyin_01 𣛙 1`] = `
[
  "shuo4",
  "shuo4",
  "shwo4",
  "shuo4",
]
`;

exports[`char2pinyin_01 𫗭 1`] = `
[
  "wei4",
  "wei4",
  "wei4",
  "wei4",
]
`;

exports[`char2pinyin_02 𠬠 1`] = `
[
  "mǒu",
  "mǒu",
  "mǒu",
  "mou3",
]
`;

exports[`char2pinyin_02 𠮷 1`] = `
[
  "jí",
  "jí",
  "jí",
  "chi2",
]
`;

exports[`char2pinyin_02 𡬶 1`] = `
[
  "xín",
  "shīn",
  "syīn",
  "hsin2",
]
`;

exports[`char2pinyin_02 𢎐 1`] = `
[
  "shàng",
  "shàng",
  "shàng",
  "shang4",
]
`;

exports[`char2pinyin_02 𣛙 1`] = `
[
  "shuò",
  "shuò",
  "shwò",
  "shuo4",
]
`;

exports[`char2pinyin_02 𫗭 1`] = `
[
  "wèi",
  "wèi",
  "wèi",
  "wei4",
]
`;
